org-roam-graph.el

```diff
@@ -144,120 +141,109 @@ into a digraph."
+                    :where (and (in to selected) (in from selected))])
            (edges-cites-query
             `[:with selected :as [:select [file] :from ,node-query]
-              :select :distinct [file from]
-              :from links :inner :join refs :on (and (= links:to refs:ref) (= links:type "cite"))
-              :where (and (in file selected) (in from selected))])
-           (edges (org-roam-db-query edges-query))
+                    :select :distinct [file from]
+                    :from links :inner :join refs :on (and (= links:to refs:ref)
+                                                           (= links:type "cite"))
+                    :where (and (in file selected) (in from selected))])
+           (edges       (org-roam-db-query edges-query))
            (edges-cites (org-roam-db-query edges-cites-query)))
       (insert "digraph \"org-roam\" {\n")
-
       (dolist (option org-roam-graph-extra-config)
-        (insert (concat "  "
-                        (car option)
-                        "="
-                        (cdr option)
-                        ";\n")))
-      (insert (format "  node [%s];\n"
-                      (->> org-roam-graph-node-extra-config
-                           (mapcar (lambda (n)
-                                     (concat (car n) "=" (cdr n))))
-                           (s-join ","))))
-      (insert (format "  edge [%s];\n"
-                      (->> org-roam-graph-edge-extra-config
-                           (mapcar (lambda (n)
-                                     (concat (car n) "=" (cdr n))))
-                           (s-join ","))))
-
+        (insert (org-roam-graph--dot-option option) ";\n"))
+      (dolist (node-type '("node" "edge"))
+        (insert (format " %s [%s];\n" node-type
+                        (mapconcat #'org-roam-graph--dot-option
+                                   (symbol-value
+                                    (intern (concat "org-roam-graph-" node-type "-extra-config")))
+                                   ","))))
       (dolist (node nodes)
         (let* ((file (xml-escape-string (car node)))
                (title (or (caadr node)
                           (org-roam--path-to-slug file)))
                (shortened-title (s-truncate org-roam-graph-max-title-length title))
-               (node-properties (list (cons "label" (s-replace "\"" "\\\"" shortened-title))
-                                      (cons "URL" (concat "org-protocol://roam-file?file="
-                                                          (url-hexify-string file)))
-                                      (cons "tooltip" (xml-escape-string title)))))
+               (node-properties
+                `(("label"   . ,(s-replace "\"" "\\\"" shortened-title))
+                  ("URL"     . ,(concat "org-protocol://roam-file?file=" (url-hexify-string file)))
+                  ("tooltip" . ,(xml-escape-string title)))))
```

Comment:
It feels like `label` should be treated the same as tooltip? (I haven't tried it out, though.)

Reply:
https://www.graphviz.org/doc/info/shapes.html#html
>  As HTML strings are processed like HTML input, any use of the ", &, <, and > characters in literal text or in attribute values need to be replaced by the corresponding escape sequence. For example, if you want to use & in an href value, this should be represented as &amp;. 

So we probably should be escaping entities.

Reply:
https://renenyffenegger.ch/notes/tools/Graphviz/attributes/label/index
>  HTML like lables
Typically, the label is just a string that is enclosed in quotes (label = "some text").
However, if angle brackes are used (label = < <table> … >), Graphviz renders labels with HTML tags.

scratch that, plain text is valid for the label and escaping would revert #314.
The HTML labels look like they could enable some interesting features by allowing us to show more information in each node, but that's for another issue.